src/views/ApartmentApp/components/StudentApplication/components/ApplicantList/components/ApplicantListItem/index.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  Grid,
+  Divider,
+  ListItem,
+  ListItemAvatar,
+  ListItemSecondaryAction,
+  ListItemText,
+  Avatar,
+  IconButton,
+} from '@material-ui/core';
+import ClearIcon from '@material-ui/icons/Clear';
+import PersonIcon from '@material-ui/icons/Person';
+import StarBorderIcon from '@material-ui/icons/StarBorder';
+import StarIcon from '@material-ui/icons/Star';
+import user from '../../../../../../../../services/user';
+
+// Based off src/views/PeopleSearch/components/PeopleSearchResult
+// but using props.profile of type StudentProfileInfo
+// rather than using this.props.Person of type PeopleSearchResult
+const ApplicantListItem = ({ profile, isApplicationEditor, onChangeEditor, onApplicantRemove }) => {
+  const [avatar, setAvatar] = useState(null);
+  const [hasNickName, setHasNickname] = useState(false);
+  const [personClass, setPersonClass] = useState(profile.Class);
+
+  useEffect(() => {
+    loadAvatar(profile);
+    createNickname(profile);
+    if (String(profile.PersonType).includes('stu') && profile.Class !== undefined) {
+      setPersonClass(profile.Class);
+    } else {
+      // Technically, this case should never happen because the list does not allow the user to add a non-student to the applicant list
```

Comment:
This is done at the top level component (StudentApplication/index.js) so that the "errors" may be shown in a snackbar or dialog box.
Way back in the early versions I tried putting that error checking in the child components but it was less reliable, inefficient, and it sort of violated the modularity of the child components